hydra-main/src/main/java/com/addthis/hydra/job/spawn/balancer/SpawnBalancer.java

```diff
@@ -420,29 +420,31 @@ public boolean canReceiveNewTasks(HostState host) {
                                                          int moveLimit,
                                                          boolean obeyDontAutobalanceMe) {
         List<JobTaskMoveAssignment> rv = purgeMisplacedTasks(host, moveLimit);
-        if (rv.size() <= moveLimit) {
-            long byteLimit = (long) (byteLimitFactor * config.getBytesMovedFullRebalance());
-            List<HostState> hostsSorted = sortHostsByDiskSpace(otherHosts);
-            for (JobTask task : findTasksToMove(host, obeyDontAutobalanceMe)) {
-                long taskTrueSize = getTaskTrueSize(task);
-                if (limitBytes && (taskTrueSize > byteLimit)) {
-                    continue;
-                }
-                JobTaskMoveAssignment assignment = moveTask(task, host.getHostUuid(), hostsSorted);
-                // we don't want to take up one of the limited rebalance slots
-                // with an assignment that we know has no chance of happening
-                // because either the assignment is null or the target host
-                // for the assignment is null
-                if (assignment != null && assignment.getTargetUUID() != null) {
-                    markRecentlyReplicatedTo(assignment.getTargetUUID());
-                    rv.add(assignment);
-                    byteLimit -= taskTrueSize;
-                }
-                if (rv.size() >= moveLimit) {
-                    break;
-                }
+        MoveAssignmentList moveAssignments = new MoveAssignmentList(spawn, taskSizer);
```

Comment:
`MoveAssignmentList.getBytesUsed` isn't used in this method, so I think we can just use a regular `List` to avoid the extra cost of the `add` method. As a matter of fact, the original code just needs minor changes:
- delete first call to `purgeMisplacedTasks`. Create of a new `List<JobTaskMveAssignment>` instead
- get rid of `if (rv.size() <= moveLimit) {` as the new list is empty
- the code within the `if` body remains unchanged
- call purgeMisplacedTasks and add its result to the list before return